Sage/carbon
# fix(dialog): add DialogSizes type to component interface (FE-5390)

closes #5516 

### Proposed behaviour

- Add missing `DialogSizes` type from the interface for the `Dialog` component
- Amend typing of `DialogSizes` so consumers can see their types as literal unions, not type alias names
- Convert story setup to CSF stories with MDX docs approach to allow for story type checking

### Current behaviour

- `DialogSizes` type is missing from `Dialog` interface
- The type of `size` prop of `Dialog` shows as `DialogSizes` and not `extra-small | small | ...`   
- Cannot type check stories

### Checklist




#### QA


### Testing instructions

- Using the provided codesandboxes, verify the following error has been resolved: ``Module '"carbon-react/lib/components/dialog"' has no exported member 'DialogSizes'.``
- In storybook:
     
     - Verify in `Dialog` prop table that the `size` prop has its possible values documented (i.e. `extra-small`, `small`, etc.) 
     - Verify all `Dialog` stories have their source code correctly rendered under `Show code` and is inline with master.
